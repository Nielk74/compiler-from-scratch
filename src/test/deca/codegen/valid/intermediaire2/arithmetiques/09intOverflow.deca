// Description:
//    Programme testant le débordement de résultats 
//    sur des opérations arithmétiques sur des entiers
//
// Resultats:
//    -2147483648
//    2147483647
//    -2
//    0
//    -2147483648
//    2147483647
//    -2147483648
//
// Historique:
//    cree le 11/06/2022

{
    int max = 2147483647;
    int min = -2147483647 - 1;

    println(max + 1);
    println(min + -1);

    println(max * 2);
    println(min * 2);

    println(max - -1);
    println(min - 1);

    println(-min);
}