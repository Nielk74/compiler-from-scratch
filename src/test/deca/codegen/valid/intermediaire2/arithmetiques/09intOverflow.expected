-2147483648
2147483647
-2
0
-2147483648
2147483647
-2147483648
